evm/estimators/gas.py

```diff
@@ -50,20 +57,36 @@ def binary_gas_search(state, transaction, tolerance=1):
         subject to tolerance. If OutOfGas is thrown at block limit, return block limit.
     :raises VMError: if the computation fails even when given the block gas_limit to complete
     """
+    if not hasattr(transaction, 'sender'):
+        raise TypeError(
+            "Transaction is missing attribute sender.",
+            "If sending an unsigned transaction, use SpoofTransaction and provide the",
+            "sender using the 'from' parameter")
+
+    transaction_context = state.get_transaction_context_class()(
+        gas_price=0,
+        origin=transaction.sender,
+        data=transaction.data,
+    )
```

Comment:
It seems like removing `intrinsic_gas()` from the transaction means we have to do a lot of setup here for the transaction_context, which is only used to get the intrinsic_gas back. What was the payoff for moving it?

I think that the intrinsic cost of the transaction should not be semantically part of the computation. It seems like the Transaction itself is the most natural place to define the intrinsic gas in a new fork. Also, what if it's a transaction with no computation at all? Would we have to build an empty computation or TransactionContext just to get the gas cost of an empty transaction?